pkg/clients/dbsubnetgroup/dbsubnetgroup.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dbsubnetgroup
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/rds"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	"github.com/crossplane/provider-aws/apis/database/v1beta1"
+	awsclients "github.com/crossplane/provider-aws/pkg/clients"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+// Client is the external client used for DBSubnetGroup Custom Resource
+type Client interface { // nolint:gocyclo
+	CreateDBSubnetGroupRequest(input *rds.CreateDBSubnetGroupInput) rds.CreateDBSubnetGroupRequest
+	DeleteDBSubnetGroupRequest(input *rds.DeleteDBSubnetGroupInput) rds.DeleteDBSubnetGroupRequest
+	DescribeDBSubnetGroupsRequest(input *rds.DescribeDBSubnetGroupsInput) rds.DescribeDBSubnetGroupsRequest
+	ModifyDBSubnetGroupRequest(input *rds.ModifyDBSubnetGroupInput) rds.ModifyDBSubnetGroupRequest
+}
+
+// NewClient returns a new client using AWS credentials as JSON encoded data.
+func NewClient(ctx context.Context, credentials []byte, region string, auth awsclients.AuthMethod) (Client, error) {
+	cfg, err := auth(ctx, credentials, awsclients.DefaultSection, region)
+	if cfg == nil {
+		return nil, err
+	}
+	return rds.New(*cfg), nil
+}
+
+// IsDBSubnetGroupNotFoundErr returns true if the error is because the item doesn't exist
+func IsDBSubnetGroupNotFoundErr(err error) bool {
+	if err == nil {
+		return false
+	}
+	return strings.Contains(err.Error(), rds.ErrCodeDBSubnetGroupNotFoundFault)
+}
+
+// IsDBSubnetGroupUpToDate checks whether there is a change in any of the modifiable fields.
+func IsDBSubnetGroupUpToDate(p v1beta1.DBSubnetGroupParameters, sg rds.DBSubnetGroup) (bool, error) {
```

Comment:
There are only 2 fields that might need update since `Modify` call takes only those. So, it might be easier just to check `DBSubnetGroupDescription` and `SubnetIDs` fields and see whether they changed and return.

Reply:
This comment assumed that we were not able to update the tags but that's actually possible later I realized. So, you need to check tags as well.